lib/cea/cea608_data_channel.js

```diff
@@ -0,0 +1,720 @@
+     */
+    this.text_ =
+        new shaka.cea.Cea608Memory(decoder, fieldNum, chanNum, true);
+
+    /**
+     * Displayed memory.
+     * @private {!shaka.cea.Cea608Memory}
+     */
+    this.dpm_ =
+        new shaka.cea.Cea608Memory(decoder, fieldNum, chanNum, false);
+
+    /**
+     * Non-displayed memory.
+     * @private {!shaka.cea.Cea608Memory}
+     */
+    this.ndm_ =
+        new shaka.cea.Cea608Memory(decoder, fieldNum, chanNum, false);
+
+    /**
+     * Points to current buffer.
+     * @public {!shaka.cea.Cea608Memory}
+     */
+    this.curbuf = this.dpm_;
+
+    /**
+     * End time of the previous caption, serves as start time of next caption.
+     * @private {!number}
+     */
+    this.prevEndTime_ = 0;
+
+    /**
+     * Last control pair, 16 bits representing byte 1 and byte 2
+     * @private {?number}
+     */
+    this.lastcp_ = null;
+  }
+
+  /**
+   * Resets channel state.
+   */
+  reset() {
+    this.type_ = shaka.cea.Cea608DataChannel.CaptionType.PAINTON;
+    this.curbuf = this.dpm_;
+    this.lastcp_ = null;
+    this.dpm_.reset();
+    this.ndm_.reset();
+    this.text_.reset();
+  }
+
+  /**
+   * Gets the row index from a Preamble Address Code byte pair.
+   * @param {!number} b1 Byte 1.
+   * @param {!number} b2 Byte 2.
+   * @return {!number} Row index.
+   * @private
+   */
+  pacToRow_(b1, b2) {
+    const ccrowtab = [
+      11, 11,  // 0x00 or 0x01
+      1, 2,    // 0x02 -> 0x03
+      3, 4,    // 0x04 -> 0x05
+      12, 13,  // 0x06 -> 0x07
+      14, 15,  // 0x08 -> 0x09
+      5, 6,    // 0x0A -> 0x0B
+      7, 8,    // 0x0C -> 0x0D
+      9, 10,   // 0x0E -> 0x0F
+    ];
+    return ccrowtab[((b1 & 0x07) << 1) | ((b2 >> 5) & 0x01)];
+  }
+
+  /**
+   * PAC - Preamble Address Code.
+   * b1 is of the form |P|0|0|1|C|0|ROW|
+   * b2 is of the form |P|1|N|ATTRIBUTE|U|
+   * @param {!number} b1 Byte 1.
+   * @param {!number} b2 Byte 2.
+   * @private
+   */
+  controlPac_(b1, b2) {
+    let row = this.pacToRow_(b1, b2);
+
+    // Get attribute bits (4 bits)
+    const attr = (b2 & 0x1E) >> 1;
+
+    // Set up the defaults.
+    let textColor = shaka.cea.Cea608Memory.DEFAULT_TXT_COLOR;
+    let italics = false;
+
+
+    // Attributes < 7 are colors, = 7 is white w/ italics, and >7 are indents
+    if (attr < 7) {
+      textColor = shaka.cea.Cea608DataChannel.TEXT_COLORS[attr];
+    } else if (attr === 7) {
+      italics = true; // color stays white
+    }
+
+    // PACs toggle underline on the last bit of b2.
+    const underline = (b2 & 0x01) === 0x01;
+
+    // Execute the PAC.
+    const buf = this.curbuf;
+    if (this.type_ === shaka.cea.Cea608DataChannel.CaptionType.TEXT) {
+      row = buf.getRow(); // Don't change row if in text mode.
+    } else if (this.type_ === shaka.cea.Cea608DataChannel.CaptionType.ROLLUP) {
+      if (row !== buf.getRow()) { // Move entire scroll window to a new base.
+        const oldtoprow = 1 + buf.getRow() - buf.getScrollSize();
```

Comment:
I had to take a moment to decide that you didn't just refer to the front of a ship ("prow").  CamelCase here would help me see that it was the oldTopRow.